hivemind/utils/asyncio.py

```diff
@@ -27,13 +34,31 @@ async def aiter(*args: T) -> AsyncIterator[T]:
         yield arg
 
 
+async def azip(*iterables: AsyncIterable[T]) -> AsyncIterator[Tuple[T, ...]]:
+    """ equivalent of zip for asynchronous iterables """
+    iterators = [iterable.__aiter__() for iterable in iterables]
+    while True:
+        try:
+            yield tuple(await asyncio.gather(*(itr.__anext__() for itr in iterators)))
+        except StopAsyncIteration:
+            break
```

Comment:
TODO maybe replace these helper functions with equivalent versions from some lightweight library and use it as a dependency

Reply:
Added to backlog for later